tools/halide_image.h

```diff
@@ -223,9 +227,14 @@ class Image {
         contents->buf.min[2] = z;
         contents->buf.min[3] = w;
     }
+
+    void info(const char *tag = "Image") const;
+    void dump(const char *tag = "Image") const;
```

Comment:
OK, I will try again on the standalone template function issue (it was probably just C++ template syntax I was getting wrong).